portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,47 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
+/** 
+ * Allows a button to be clicked successively in order to
+ * display a personal summary, then a photograph. 
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function showNext() {
+  const summaryContainer = document.getElementById('summary');
+  const imageContainer = document.getElementById('picture');
+  if (summaryContainer.innerText.length == 0) {
+    showSummary();
+  } else if (imageContainer.childElementCount == 0) { 
+    showPicture();
+  }
+}
+
+function showSummary(){
+  const summaryContainer = document.getElementById('summary');
+  const summaryText = 'Currently, I\'m a 2nd year student' + 
+  ' at the Rose-Hulman Institute of Technology working towards' +
+  ' a bachelor in Computer Science. I grew up in Louisville,' +
+  ' Kentucky and moved on to college early.I completed two years' +
+  ' at Western Kentucky University while simultaneously obtaining' +
+  ' my highschool diploma. The summer of my graduation I was' +
+  ' accepted as a Google CSSI student and spent 3 weeks at the' +
+  ' Google Cambridge Office. A year later and I am now working' +
+  ' as a virtual Google intern in my hometown of Louisville.';
+  summaryContainer.innerText = summaryText;
+}
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+function showPicture(){
+  const imgUrl = 'Me.jpg';
+  const imgElement = document.createElement('img');
+  imgElement.width = '550';
+  imgElement.height = '450';
+  imgElement.src = imgUrl;
+  const imageContainer = document.getElementById('picture');
+  imageContainer.appendChild(imgElement);    
+}
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+function getServerData(){
+  fetch('/data').then(response => response.json()).then((messages) => {
+    const serverContainer = document.getElementById('server-container');
+    serverContainer.innerText = messages[Math.floor(Math.random()*5)];
```

Comment:
This is another place that depends on the length of `messages` always being 5. It would be very easy for DataServlet.java to change and accidentally break this line.